Sources/ClientRuntime/Networking/Streaming/DataStreamReader.swift

```diff
@@ -42,15 +42,21 @@ public class DataStreamReader: StreamReader {
     }
     
     public func read(maxBytes: UInt? = nil, rewind: Bool = false) -> ByteBuffer {
-        let buffer = ByteBuffer(size: Int(maxBytes ?? availableForRead))
+        var data = Data()
         withLockingClosure {
-            buffer.put(byteBuffer, offset: offset, maxBytes: maxBytes)
-            if !rewind {
-                _availableForRead -= UInt(buffer.length)
-                offset += UInt(buffer.length)
+            let count = Int(maxBytes ?? availableForRead)
+            let ptr = UnsafeMutablePointer<UInt8>.allocate(capacity: count)
+            let buffer = UnsafeMutableBufferPointer(start: ptr, count: count)
+            let bytesRead = byteBuffer.read(buffer: buffer)
+            data = Data(buffer)
+            ptr.deallocate()
+                        
+            if !rewind, let bytesRead = bytesRead {
+                _availableForRead -= UInt(truncatingIfNeeded: bytesRead)
```

Comment:
shouldn't it fail if truncation happens?

This could lead to loss of data.

Reply:
Yeah good call out, I'll remove the `truncatingIfNeeded`. 
When we tackle streaming improvements, we may want to update the interface to use `UIInt64` as well